### PR TITLE
System version upgrade

### PR DESCRIPTION
**Pull request by cvetkovic**
Upgrade of system components to:
- Kubernetes: 1.23.5
- Knative: 1.4
- Istio: 1.12.5
- containerd: 1.6.2
- protoc: 3.19.4
- runc: 1.1.0
- Go: 1.18

### CODE REVIEW

File: scripts/cluster/create_multinode_cluster.sh [hunk 1]
```diff
@@ -48,4 +50,4 @@ while true; do
     esac
 done
 
-$DIR/setup_master_node.sh
\ No newline at end of file
+$DIR/setup_master_node.sh $STOCK_CONTAINERD
```
Comment (HongyuHe) [inline scripts/cluster/create_multinode_cluster.sh:53]: Just to clarify. I need this to set up stock clusters.

**Review by HongyuHe — APPROVED** [review]
Thanks for upgrading knative!

File: scripts/cluster/create_multinode_cluster.sh (hunk 1 above)
Comment (cvetkovic) [reply scripts/cluster/create_multinode_cluster.sh:53]: Okay, I did rebase, so that's why the commit tree is different.

**Review by ustiugov — CHANGES_REQUESTED** [review]
Looks good, thanks. Please go over the workflow files and upgrade Go version. Also, upgrade the dependency versions in the developer guide.

Have you opened a new PR in our knative serving fork? If not please do.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@ jobs:
   build:
     strategy:
       matrix:
-        go: ['1.13', '1.14', '1.15', '1.16']
+        go: ['1.13', '1.14', '1.15', '1.16', '1.17', '1.18']
```
Comment (ustiugov) [inline .github/workflows/build.yml:25]:
```suggestion
        go: ['1.16', '1.17', '1.18']
```

File: docs/developers_guide.md [hunk 3]
```diff
@@ -183,7 +183,7 @@ Checkout [this](https://www.scalyr.com/blog/zipkin-tutorial-distributed-tracing/
 of our [fork](https://github.com/ease-lab/firecracker-containerd) of the upstream repository.
 Currently, we are in the process of upstreaming VM snapshots support to the upstream repository.
 
-* Current Firecracker version is 0.24.0, Knative 0.26.0, Kubernetes 1.20.6, gVisor 20210622.0, and Istio 1.9.5.
+* Current Firecracker version is 0.24.0, Knative 1.3, Kubernetes 1.23.5, gVisor 20210622.0, and Istio 1.12.5.
```
Comment (ustiugov) [inline docs/developers_guide.md:186]: It seems you also updated gVisor in `install_stock.sh` to 20220117. Please check if it's the latest version and if the tests pass

**Review by ustiugov — CHANGES_REQUESTED** [review]
Minor comments and please update the changelog in the unreleased version section.

**Comment by cvetkovic** [comment]
I think that CRI tests fail because runner containers have preinstalled Kubernetes v1.20.15, which is not compatible with the upgraded versions of Knative (v1.3) and Istio (v1.12.5). 

Also, I tried running tests manually inside a container but wasn't able to reproduce the error. Instead, I encountered some other problems (see attachment).

@shyamjesal Could you have a look and upgrade Kubernetes for CRI tests?

Screenshot from 2022-04-01 12-17-28

**Comment by ustiugov** [comment]
hi @cvetkovic, shall we merge this one? it needs to be rebased though

File: .github/workflows/integration_tests.yml [hunk 4]
```diff
@@ -28,10 +28,11 @@ jobs:
       matrix:
         test-name: [test, test-man-bench]
     steps:
-    - name: Set up Go 1.16
-      uses: actions/setup-go@v3
+    
+    - name: Set up Go 1.18
+      uses: actions/setup-go@v2
```
Comment (ustiugov) [inline .github/workflows/integration_tests.yml:33]:
```suggestion
      uses: actions/setup-go@v3
```

**Comment by ustiugov** [comment]
@cvetkovic please squash last 2 commits using `git rebase -i`, the last commit is too minor. fewer commits make the repo commit history easier to navigate

**Review by ustiugov — APPROVED** [review]
thanks! let now wait for @shyamjesal to fix the runner

**Comment by shyamjesal** [comment]
Updated the runners to kind version v1.23.5. CRI tests are passing too. @cvetkovic can you please upgrade the knative version to the latest one and then we can merge the PR finally.

**Review by ustiugov — APPROVED** [review]

**Pull request merged**